pintracker/stateless/stateless_test.go

```diff
@@ -0,0 +1,532 @@
+package stateless
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	rpc "github.com/hsanjuan/go-libp2p-gorpc"
+	cid "github.com/ipfs/go-cid"
+	peer "github.com/libp2p/go-libp2p-peer"
+	"sort"
+	"strings"
+	"testing"
+	"time"
+
+	ipfscluster "github.com/ipfs/ipfs-cluster"
+	"github.com/ipfs/ipfs-cluster/api"
+	"github.com/ipfs/ipfs-cluster/test"
+)
+
+var (
+	logLevel               = "CRITICAL"
+	customLogLvlFacilities = logFacilities{}
+	pinCancelCid           = test.TestCid3
+	unpinCancelCid         = test.TestCid2
+	ErrPinCancelCid        = errors.New("should not have received rpc.IPFSPin operation")
+	ErrUnpinCancelCid      = errors.New("should not have received rpc.IPFSUnpin operation")
+)
+
+type logFacilities []string
+
+// String is the method to format the flag's value, part of the flag.Value interface.
+func (lg *logFacilities) String() string {
+	return fmt.Sprint(*lg)
+}
+
+// Set is the method to set the flag value, part of the flag.Value interface.
+func (lg *logFacilities) Set(value string) error {
+	if len(*lg) > 0 {
+		return errors.New("logFacilities flag already set")
+	}
+	for _, lf := range strings.Split(value, ",") {
+		*lg = append(*lg, lf)
+	}
+	return nil
+}
+
+func init() {
+	flag.Var(&customLogLvlFacilities, "logfacs", "use -logLevel for only the following log facilities; comma-separated")
+	flag.StringVar(&logLevel, "loglevel", logLevel, "default log level for tests")
+	flag.Parse()
+
+	if len(customLogLvlFacilities) <= 0 {
+		for f := range ipfscluster.LoggingFacilities {
```

Comment:
This introduces a dependency back to cluster from this package. 

As a policy, components should not need to use the `ipfscluster` package for anything as they are fully independent and this can only complicate things in the future (create cycles, difficult extractions etc).


This also introduces code which is essentially a dup from the main package tests. My impression is that, since these tests are limited to this component you should not need any fine-tuning of the facilities (there is basically 1 only).

Reply:
I want to split this test logging flags into a separate package, in a different PR, I will just remove it from here for now. Was only for debugging purposes anyway.